### DISCUSSION TITLE
Feature request | host:port list

### DISCUSSION THREAD

**Discussion by mr-rizwan-syed** [projectdiscovery/httpx#789] [Q&A] [answered]
Feature request
 -l, -list string      input file containing list of hosts and port to process

list containg host with port probe

127.0.0.1:8080
sd.tesla.com:10000
google.com:80

Only scans port which are open no need to specify explicitly

httpx -l $subdomains -p $naabuports    < this will scan all the ports identified on all host - not efficient

This would be better if were able to supply file containg a list of host:port

**Comment by olearycrew** [comment, marked as answer]
This worked for me:

file `withport`

<img width="590" alt="CleanShot 2023-04-19 at 12 01 40@2x" src="https://user-images.githubusercontent.com/6044920/233133771-8333de69-008e-4fe4-b3f3-9287c56d02f6.png">

`httpx -l withport`

<img width="769" alt="CleanShot 2023-04-19 at 12 01 57@2x" src="https://user-images.githubusercontent.com/6044920/233133849-fe12954b-3943-4cba-82e5-aebe5782f589.png">

(host didn't respond on 1234 but did without a port and with 443)
